internal/model/db.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 The golang.design Initiative Authors.
+// All rights reserved. Use of this source code is governed
+// by a MIT license that can be found in the LICENSE file.
+//
+// Originally written by Changkun Ou <changkun.de> at
+// changkun.de/s/redir, adopted by Mai Yang <maiyang.me>.
+
+package model
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+var (
+	ErrExistedAlias = errors.New("alias is existed")
+)
+
+type AliasKind int
+
+const (
+	KindShort AliasKind = iota
+	KindRandom
+)
+
+const (
+	dbname = "redir"
+)
+
+// Redirect records a kind of alias and its correlated link.
+type Redirect struct {
+	Alias   string    `json:"alias"   db:"alias"`
+	Kind    AliasKind `json:"kind"    db:"kind"`
+	URL     string    `json:"url"     db:"url"`
+	Private bool      `json:"private" db:"private"`
+}
+
+// Visit indicates an Record of Visit pattern.
+type Visit struct {
+	Alias   string    `json:"alias"   db:"alias"`
+	Kind    AliasKind `json:"kind"    db:"kind"`
+	IP      string    `json:"ip"      db:"ip"`
+	UA      string    `json:"ua"      db:"ua"`
+	Referer string    `json:"referer" db:"referer"`
+	Time    time.Time `json:"time"    db:"time"`
+}
+
+type Refstat struct {
+	Referer string `json:"referer" db:"referer"`
+	Count   int64  `json:"count"   db:"count"`
+}
+
+type UAstat struct {
+	UA    string `json:"ua"    db:"ua"`
+	Count int64  `json:"count" db:"count"`
+}
+
+type Locstat struct {
+	Locations []string `db:"locs" json:"locs"`
+}
+
+type Timehist struct {
+	Time  time.Time `db:"time"  json:"time"`
+	Count int       `db:"count" json:"count"`
+}
```

Comment:
stat 相关的数据属于视图，只会返回给前端而不会存储到 db 中，应该将不需要的 tag 删掉

```suggestion
type Refstat struct {
	Referer string `json:"referer"`
	Count   int64  `json:"count"`
}

type UAstat struct {
	UA    string `json:"ua"`
	Count int64  `json:"count"`
}

type Timehist struct {
	Time  time.Time `json:"time"`
	Count int       `json:"count"`
}
```